handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1606,8 +1607,8 @@ private boolean isDestroyed() {
         return destroyed != 0;
     }
 
-    static int calculateOutNetBufSize(int pendingBytes) {
-        return min(MAX_ENCRYPTED_PACKET_LENGTH, MAX_ENCRYPTION_OVERHEAD_LENGTH
+    static int calculateOutNetBufSize(int pendingBytes, int numComponents) {
+        return min(MAX_ENCRYPTED_PACKET_LENGTH, MAX_ENCRYPTION_OVERHEAD_LENGTH * numComponents
```

Comment:
This is described in the commit message but if we have a `CompositeByteBuf` we break that down into `ByteBuffer[]` and call `SSL_write` for each of these. So each `ByteBuffer` will incur the tls record overhead. I wanted to review this more because it may be the case that we can apply the encryption overhead to all the bytes, and other apply the tls record overhead (header) for each `ByteBuffer`. I think this case ... so let me refine this estimate.